test/check-machines-migrate

```diff
@@ -0,0 +1,187 @@
+#!/usr/bin/python3
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2021 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import sys
+
+# import Cockpit's machinery for test VMs and its browser test API
+TEST_DIR = os.path.dirname(__file__)
+sys.path.append(os.path.join(TEST_DIR, "common"))
+sys.path.append(os.path.join(os.path.dirname(TEST_DIR), "bots/machine"))
+
+from machineslib import VirtualMachinesCase  # noqa
+from testlib import wait, test_main  # noqa
+from machinesxmls import POOL_XML  # noqa
+
+
+class TestMachinesMigration(VirtualMachinesCase):
+    provision = {
+        "machine1": {"address": "10.111.113.1/20", "dns": "10.111.112.100"},
+        "machine2": {"address": "10.111.113.2/20", "dns": "10.111.112.100"},
+    }
+
+    def prepare(self, machine1, machine2):
+        machine1.execute("echo '10.111.113.2 machine2' >> /etc/hosts")
+        machine1.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.2 machine2' >> /etc/hosts")
+        machine2.execute("echo '10.111.113.1 machine1' >> /etc/hosts")
+
+        machine2.execute("sed -i 's/127.0.1.1.*/127.0.1.1 machine2 machine2/' /etc/hosts")
```

Comment:
I thought the /etc/hosts entries are like: 127.0.1.1 machines2.example.com machine2 where the first is a DNS name and the second is just the hostname.

I figure out I can just remove the first since it does not really exist.